src/Rings/localization_interface.jl

```diff
@@ -0,0 +1,361 @@
+export AbsMultSet, AbsPowersOfElement, AbsComplementOfPrimeIdeal
+
+export AbsLocalizedRing
+export original_ring, inverted_set
+export reduce_fraction
+export localize_at
+
+export AbsLocalizedRingElem
+export numerator, denominator, parent
+
+export AbsLocalizedIdeal
+export ideal
+
+import AbstractAlgebra.Ring
+
+#################################################################################
+# General framework for localizations of rings; to be used with affine algebras #
+#################################################################################
+
+#################################################################################
+# Multiplicatively closed sets of (commutative) rings                           #
+#################################################################################
+
+@Markdown.doc """
+    AbsMultSet{RingType, RingElemType}
+
+Abstract type for a multiplicatively closed set in a commutative (Noetherian) ring 
+R of type `RingType` with elements of type `RingElemType`.
+"""
+abstract type AbsMultSet{RingType, RingElemType} end
```

Comment:
Perhaps the type parameters should be restricted? E.g.
```suggestion
abstract type AbsMultSet{RingType<:Ring, RingElemType<:RingElem} end
```
or
```suggestion
abstract type AbsMultSet{RingType<:Ring, RingElemType<:RingElement} end
```